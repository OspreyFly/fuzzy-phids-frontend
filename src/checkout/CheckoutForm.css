.checkout-form {
    background-color: #008170; /* Light grey background */
    padding: 20px;
    border-radius: 10px;
    box-shadow: 0 4px 8px rgba(0,91,65,0.5);
    margin-top: 20px; /* Space from the last item */
    text-align: center; /* Center text */
    border: white solid 1px;
  }
  
  .checkout-form input {
    background-color: #ffffff; /* Blue background */
    font-size:30px;
    border: none;
    padding: 10px 20px;
    border-radius: 5px;
    cursor: pointer;
    transition: background-color 0.3s ease;
    margin: 5px;
  }
  
  .checkout-form input[type="submit"]:hover {
    background-color: #0056b3; /* Darker blue on hover */
  }

p{
  color: white;
}