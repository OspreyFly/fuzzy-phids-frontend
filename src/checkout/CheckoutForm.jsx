import React, { useState, useContext, useEffect } from 'react';
import UserContext from '../auth/UserContext';
import FuzzyApi from '../api/FuzzyApi';
import './CheckoutForm.css';

const CheckoutForm = ({ cart, setCart }) => {
    const { currentUser } = useContext(UserContext);
    const [showSuccessPopup, setShowSuccessPopup] = useState(false);
    const [popupMessage, setPopupMessage] = useState('');
    const [formData, setFormData] = useState({
        phone: '',
        delivery_address: '',
        items: cart,
        user_order_id: currentUser.id
    });

    useEffect(() => {
        // Calculate the total after the component has mounted
        const total = getTotal();
        setFormData(prevFormData => ({
            ...prevFormData,
            total: total,
        }));
    }, [cart]); // Empty dependency array means this effect runs once after mounting

    function getTotal() {
        let total = 0;
        for (let item of cart) {
            total += +item.price;
        }
        return total;
    }

    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData({ ...formData, [name]: value });
    };

    const handleSubmit = async (e) => {
        e.preventDefault();
        // Perform validation here
        if (isValid(formData)) {
            try {
                console.log('Form is valid, proceed with submission.');
                const res = await FuzzyApi.createOrder({ ...formData });
                setShowSuccessPopup(true);
                setPopupMessage('Your order was submitted successfully!');
                setCart([]);
            } catch (e) {
                console.error(e);
                setShowSuccessPopup(true);
                setPopupMessage('Your order was not submitted!');
            }

        } else {
            console.error('Form validation failed.');
        }
    };

    const isValid = (data) => {
        if (data.phone && data.delivery_address) {
            return true;
        }


    };

    return (
        <>
            <form className="checkout-form" onSubmit={handleSubmit}>
                {/* Phone */}
                <div>
                    <label htmlFor="phone">Phone:</label>
                    <input className="checkout-form"
                        type="text"
                        id="phone"
                        name="phone"
                        value={formData.phone}
                        onChange={handleChange}
                        minLength="1"
                        maxLength="30"
                        required
                    />
                </div>

                {/* Delivery Address */}
                <div>
                    <label htmlFor="delivery_address">Delivery Address:</label>
                    <input
                        type="text"
                        id="delivery_address"
                        name="delivery_address"
                        value={formData.delivery_address}
                        onChange={handleChange}
                        minLength="1"
                        maxLength="10"
                        required
                    />
                </div>

                <button type="submit">Submit</button>
            </form>
            {showSuccessPopup && (
                <div className="success-popup">
                    <p>{popupMessage}</p>
                    <button onClick={() => setShowSuccessPopup(false)}>Close</button>
                </div>
            )}
        </>
    );
};

export default CheckoutForm;
