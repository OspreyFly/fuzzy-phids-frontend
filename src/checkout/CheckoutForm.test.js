import React from 'react';
import { render } from '@testing-library/react';
import CheckoutForm from './CheckoutForm';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));
const currentUser = {id: 1, username: "testuser"};
const cart = [{id:1, species: "a", price: 1}, {id:2, species: "b", price: 1}];

describe('CheckoutForm', () => {
  test('renders correctly on initial load', () => {
    const { asFragment } = render(
        <CheckoutForm currentUser={currentUser} cart={cart} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});


