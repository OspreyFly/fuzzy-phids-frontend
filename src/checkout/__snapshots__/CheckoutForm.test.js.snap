// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutForm renders correctly on initial load 1`] = `
<DocumentFragment>
  <form
    class="checkout-form"
  >
    <div>
      <label
        for="phone"
      >
        Phone:
      </label>
      <input
        class="checkout-form"
        id="phone"
        maxlength="30"
        minlength="1"
        name="phone"
        required=""
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="delivery_address"
      >
        Delivery Address:
      </label>
      <input
        id="delivery_address"
        maxlength="10"
        minlength="1"
        name="delivery_address"
        required=""
        type="text"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
